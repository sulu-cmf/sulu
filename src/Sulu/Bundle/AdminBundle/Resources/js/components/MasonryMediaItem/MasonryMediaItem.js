// @flow
import type {Node} from 'react';
import React from 'react';
import classNames from 'classnames';
import Checkbox from '../Checkbox';
import Icon from '../Icon';
import type {MasonryItem as MasonryItemProps} from '../Masonry/types';
import masonryMediaItemStyles from './masonryMediaItem.scss';

type Props = MasonryItemProps & {
    children: Node[],
    /** The title which will be displayed in the header besides the checkbox */
    mediaTitle: string,
    /** For setting meta information like the file size or extension  */
    metaInfo?: string,
    /** The icon used inside the media overlay */
    icon?: string,
};

export default class MasonryMediaItem extends React.PureComponent<Props> {
    static defaultProps = {
        selected: false,
    };

    handleClick = () => {
        const {id} = this.props;

        if (this.props.onClick) {
            this.props.onClick(id);
        }
    };

    handleSelectionChange = (checked: boolean, id?: string | number) => {
        if (this.props.onSelectionChange && id) {
            this.props.onSelectionChange(id, checked);
        }
    };

    handleHeaderClick = () => {
        const {id, selected} = this.props;

        this.handleSelectionChange(!selected, id);
    };

    render() {
        const {
            id,
            icon,
            selected,
            metaInfo,
            children,
            mediaTitle,
        } = this.props;
        const masonryClass = classNames(
            masonryMediaItemStyles.container,
            {
                [masonryMediaItemStyles.selected]: selected,
            }
        );

        return (
            <div className={masonryClass}>
                <div className={masonryMediaItemStyles.header}>
                    <div
                        className={masonryMediaItemStyles.headerClickArea}
                        onClick={this.handleHeaderClick} />
                    <div className={masonryMediaItemStyles.mediaTitle}>
                        <Checkbox
                            value={id}
                            checked={selected}
                            onChange={this.handleSelectionChange}
                            className={masonryMediaItemStyles.checkbox}>
                            <div className={masonryMediaItemStyles.mediaTitleText}>
                                {mediaTitle}
                            </div>
                        </Checkbox>
                    </div>
                    <div className={masonryMediaItemStyles.metaInfo}>
                        {metaInfo}
                    </div>
                </div>
                <div
                    className={masonryMediaItemStyles.media}
                    onClick={this.handleClick}>
                    {children}
                    <div className={masonryMediaItemStyles.mediaOverlay}>
                        {!!icon &&
                            <Icon name={icon} className={masonryMediaItemStyles.mediaIcon} />
                        }
                    </div>
                </div>
            </div>
        );
    }
}
