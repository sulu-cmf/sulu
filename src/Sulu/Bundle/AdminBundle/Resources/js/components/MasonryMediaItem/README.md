```
<MasonryMediaItem
    metaInfo="image/png, 3,2 MB"
    mediaTitle="Massive Art Dominik Khuny">
    <img src="http://lorempixel.com/300/100"/>
</MasonryMediaItem>
```
