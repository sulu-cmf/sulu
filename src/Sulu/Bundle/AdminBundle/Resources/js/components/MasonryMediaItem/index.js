// @flow
import MasonryMediaItem from './MasonryMediaItem.js';

export default MasonryMediaItem;
