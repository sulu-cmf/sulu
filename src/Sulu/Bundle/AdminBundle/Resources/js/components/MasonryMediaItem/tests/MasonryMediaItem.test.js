/* eslint-disable flowtype/require-valid-file-annotation */
import {mount, render} from 'enzyme';
import React from 'react';
import MasonryMediaItem from '../MasonryMediaItem';

test('Render a simple MasonryMediaItem component', () => {
    expect(render(
        <MasonryMediaItem
            mediaTitle="Test"
            metaInfo="Test/Test">
            <img src="http://lorempixel.com/300/200" />
        </MasonryMediaItem>
    )).toMatchSnapshot();
});

test('Clicking on an item should call the responsible handler on the MasonryMediaItem component', () => {
    const clickSpy = jest.fn();
    const selectionSpy = jest.fn();
    const itemId = 'test';

    const masonry = mount(
        <MasonryMediaItem
            id={itemId}
            mediaTitle="Test"
            metaInfo="Test/Test"
            onClick={clickSpy}
            onSelectionChange={selectionSpy}>
            <img src="http://lorempixel.com/300/200" />
        </MasonryMediaItem>
    );

    masonry.find('MasonryMediaItem .media').simulate('click');
    expect(clickSpy).toHaveBeenCalledWith(itemId);

    masonry.find('MasonryMediaItem .headerClickArea').simulate('click');
    expect(selectionSpy).toHaveBeenCalledWith(itemId, true);
});
