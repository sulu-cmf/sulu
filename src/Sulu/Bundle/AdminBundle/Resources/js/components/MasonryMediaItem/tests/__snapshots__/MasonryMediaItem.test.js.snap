// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a simple MasonryMediaItem component 1`] = `
<div
  class="container"
>
  <div
    class="header"
  >
    <div
      class="headerClickArea"
    />
    <div
      class="mediaTitle"
    >
      <label
        class="label"
      >
        <span
          class="switch checkbox dark checkbox"
        >
          <input
            type="checkbox"
          />
          <span />
        </span>
        <div>
          <div
            class="mediaTitleText"
          >
            Test
          </div>
        </div>
      </label>
    </div>
    <div
      class="metaInfo"
    >
      Test/Test
    </div>
  </div>
  <div
    class="media"
  >
    <img
      src="http://lorempixel.com/300/200"
    />
    <div
      class="mediaOverlay"
    />
  </div>
</div>
`;
